### PR TITLE
Bug fix: Dynamodb update_table missing table entries in output

### PR DESCRIPTION
**Pull request by IreneLime**
## Motivation
This PR resolves the issue: https://github.com/localstack/localstack/issues/11781
Description of issue: The output from the UpdateTable command does not include the SSEDescription for a table which was created with it.

## Previous Behaviour
As described by the previous comment on line 781 of localstack-core\localstack\services\dynamodb\provider.py, **DynamoDBLocal cannot change certain table parameters**, including SSEDescription and replica.
As a result, the return value of update_table is a **table returned by DynamoDBLocal, which does not contain these parameters**.

## Changes
After DynamoDBLocal makes changes to the table, use get_table_schema to **retrieve the entire schema definition of the table, which includes the original and changed parameters.** 
Change return value such that the update_table function returns the schema obtained.

## Testing
Create a dynamodb table:
`awslocal dynamodb create-table --table-name Example --attribute-definitions AttributeName=key,AttributeType=S  --key-schema AttributeName=key,KeyType=HASH  --provisioned-throughput ReadCapacityUnits=5,WriteCapacityUnits=5  --sse-specification Enabled=true,SSEType=KMS,KMSMasterKeyId=some-kms`

Update the dynamodb table:
`awslocal dynamodb update-table --table-name Example --billing-mode PAY_PER_REQUEST`

Original code will not return the original SSEDescription. New changes will include both SSEDescription and the changed "Billing Mode" parameters.

### CODE REVIEW

**Comment by IreneLime** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by IreneLime** [comment]
I do not have the permission to add labels to the PR. Please add a `semver: patch` label to this PR. Thanks!

**Comment by viren-nadkarni** [comment]
Thank you for contributing @IreneLime. Could you please add some tests?

**Comment by IreneLime** [comment]
Hi @viren-nadkarni, I have added an aws service test on dynamodb for this case.
**Test added**: test_dynamodb_update_table_without_sse_specification_change will create a table with SSESpecification, and update other contents in the table, then check if "SSESpecification" entries still exist in the table.

**To run the test**: `TEST_PATH="tests/aws/services/dynamodb/test_dynamodb.py::TestDynamoDB::test_dynamodb_update_table_without_sse_specification_change" make test`


**Without the fix**: Test will fail because "SSESpecification" does not exist in returned table content.
**With the fix**: Test passes with matching "SSESpecification" content.

**Review by viren-nadkarni — COMMENTED** [review]
Great job adding the test! Just one hiccup:

File: tests/aws/services/dynamodb/test_dynamodb.py [hunk 1]
```diff
@@ -1653,6 +1653,39 @@ def test_dynamodb_create_table_with_partial_sse_specification(
         result = aws_client.dynamodb.describe_table(TableName=table_name)
         assert "SSESpecification" not in result["Table"]
 
+    @markers.aws.validated
```
[the remark this answers is not in the document]
Comment (IreneLime) [reply tests/aws/services/dynamodb/test_dynamodb.py:1656]: Thank you for the review. I have converted it into a parity test in the latest commits and tested it against AWS setup to ensure that SSEDescription related entries from the output of update_table remain consistent.

**Review by viren-nadkarni — APPROVED** [review]
Congrats on your first contribution to LocalStack!

**Pull request merged**